RFCs/0000-expand-new-task.md

```diff
@@ -0,0 +1,164 @@
+
+- Users should be able to more easily author "cross-platform" and "shared implementation" tasks.
+
+## Proposed Implementation Design
+
+### Add `--language` option to `pdk new task`
+
+- Extend the current Bolt Task object templates to support language-specific templates.
+
+- Add templates for Bash, PowerShell, Ruby, and Python.
+
+- Ruby and Python templates should have scaffold implementations based on those languages Bolt Task helper libraries. (https://github.com/puppetlabs/puppetlabs-ruby_task_helper and https://github.com/puppetlabs/puppetlabs-python_task_helper)
+
+- Ideally, users could configure a default task language as a module-specific configuration option.
+
+- The new `--language` option will require an argument which is the name of the language-specific task template to be used.
+
+- Acceptable values for the `--language` option will initially be `bash`, `powershell`, `ruby`, and `python`.
+
+- The format of future language option values should be a concise, unique, all lowercase language identifier.
+
+- If an unrecognized language option value is specified, PDK will exit with an error.
+
+- If the `--language` option is specified without an argument, PDK will exit with an error.
+
+#### Examples:
+
+```bash
+$ pdk new task my_default_task
+pdk (INFO): Using default task language: 'bash'
+pdk (INFO): Creating '/Users/jesse/sandbox/pdk/testmod/tasks/my_default_task.sh' from template.
+pdk (INFO): Creating '/Users/jesse/sandbox/pdk/testmod/tasks/my_default_task.json' from template.
+```
+
+```bash
+$ pdk new task --language=ruby my_ruby_task
+pdk (INFO): Creating '/Users/jesse/sandbox/pdk/testmod/tasks/my_ruby_task.rb' from template.
+pdk (INFO): Creating '/Users/jesse/sandbox/pdk/testmod/tasks/my_ruby_task.json' from template.
+```
+
+```bash
+$ pdk new task --language=powershell my_ps_task
+pdk (INFO): Creating '/Users/jesse/sandbox/pdk/testmod/tasks/my_ps_task.ps1' from template.
+pdk (INFO): Creating '/Users/jesse/sandbox/pdk/testmod/tasks/my_ps_task.json' from template.
+```
+
+### Add `--from` option to `pdk new task`
+
+- The new `--from` option will require an argument which is an absolute or relative path to an existing script file.
+
+- If the `--from` option is specified without an argument, PDK will exit with an error.
+
+- If the given file does not exist or is not readable, PDK will exit with an error.
+
+- If the given file exists and is readable, PDK will copy it into the `tasks` subdirectory of the current module and write the copy with a name matching the task name supplied to `pdk new task`.
+
+  - For example: `pdk new task --from=/tmp/myscript.sh restart` will copy `/tmp/myscript.sh` to `tasks/restart.sh`.
+
+- For languages that support it (currently only PowerShell?), PDK will attempt to automatically determine the parameters (and output format?) of the source script and use that data to automatically populate the Bolt Task metadata file.
```

Comment:
output schemas got dropped from the task spec before the initial release. Is that what this was thinking about?

Reply:
Sounds like we can drop "(and output format?)" in that case. Theoretically we could still populate the input parameters based on Powershell AST.